File: ios/SwiftKotlination/SwiftKotlinationTests/Sources/Features/TopStories/Factory/Mocks/TopStoriesFactoryMock.swift
@testable import SwiftKotlination

struct TopStoriesFactoryMock {
    let topStoriesManager: TopStoriesManagerMock
    let imageManager: ImageManagerMock

    init(topStoriesManager: TopStoriesManagerMock = TopStoriesManagerMock(), imageManager: ImageManagerMock = ImageManagerMock()) {
        self.topStoriesManager = topStoriesManager
        self.imageManager = imageManager
    }
}

extension TopStoriesFactoryMock: TopStoriesFactory {
    func makeTopStoriesTableViewController() -> TopStoriesTableViewController {
        let viewController = TopStoriesTableViewController.storyBoardInstance
        viewController.viewModel = TopStoriesViewModel(factory: self)

        return viewController
    }

    func makeTopStoriesManager() -> TopStoriesManagerProtocol {
        return topStoriesManager
    }
}

extension TopStoriesFactoryMock: ImageFactory {
    func makeImageManager() -> ImageManagerProtocol {
        return imageManager
    }
}

File: android/SwiftKotlination/features/topstories/src/test/kotlin/fr/jhandguy/topstories/factory/mocks/TopStoriesFactoryMock.kt
package fr.jhandguy.topstories.factory.mocks

import android.app.Activity
import fr.jhandguy.image.model.ImageManagerInterface
import fr.jhandguy.topstories.coordinator.TopStoriesCoordinatorInterface
import fr.jhandguy.topstories.factory.TopStoriesFactory
import fr.jhandguy.topstories.model.TopStoriesManagerInterface
import fr.jhandguy.topstories.viewmodel.TopStoriesViewModel

class TopStoriesFactoryMock(
    val topStoriesManager: TopStoriesManagerInterface,
    val imageManager: ImageManagerInterface,
    val topStoriesCoordinator: TopStoriesCoordinatorInterface
) : TopStoriesFactory {
    override fun makeTopStoriesManager(): TopStoriesManagerInterface = topStoriesManager
    override fun makeTopStoriesViewModel(): TopStoriesViewModel = TopStoriesViewModel(this)
    override fun makeTopStoriesCoordinator(activity: Activity): TopStoriesCoordinatorInterface = topStoriesCoordinator
    override fun makeImageManager(): ImageManagerInterface = imageManager
}
